### PR TITLE
[PP][ops-11605] NPS ReferenceCheck API

### PR DESCRIPTION
**Pull request by pawelpanasewicz**
- [PP][OPS-11605] improve check-reference API wip
- [PP][OPS-11605] finally tests compile
- [PP][OPS-11605] fixed tests

### CODE REVIEW

File: app/nps/NpsConnector.scala [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        val encoder: Base64.Encoder = Base64.getEncoder
+        val encodedBytes = encoder.encode(input.getBytes("UTF-8"))
+        new String(encodedBytes, "UTF-8")
+      }
+    val credentials = s"${appConfig.Nps.username}:${appConfig.Nps.password}"
+    val credentialsEncoded = encodeString(credentials)
+    "Authorization" -> s"Basic $credentialsEncoded"
+  }
+
+  private def makeCorrelationIdHeader(): (String, String) = {
```
Comment (barnesjake) [inline app/nps/NpsConnector.scala:114]: I wonder if we should compute this similar to something like JourneyId and store within journey throughout

**Review by barnesjake — APPROVED** [review]

File: app/nps/NpsConnector.scala (hunk 1 above)
Comment (pawelpanasewicz) [reply app/nps/NpsConnector.scala:114]: It has to be unique. However we could log it along with the journeyId or use it in audit events.

Comment (barnesjake) [reply app/nps/NpsConnector.scala:114]: https://github.com/hmrc/essttp-backend/blob/main/app/services/CorrelationIdGenerator.scala
https://github.com/hmrc/essttp-backend/blob/b83748c84d829447ef796e4c9ac228b55e4eabca/app/services/JourneyFactory.scala
Sorry I mean like here, we had to introduce a correlation id value when talking to ttp/IF

Comment (barnesjake) [reply app/nps/NpsConnector.scala:114]: I think correlation id needs to be consistent between all calls through journey for traceability across all systems e2e, but I may be misunderstanding our requirement in this project

**Pull request merged**